### PR TITLE
Resolve null-dereference and enable Linux containers

### PR DESCRIPTION
**Pull request by davidlt**
Notice: commit messages contains more information.

CMSSW fails on at least one site where Linux containers are used (probably instead of virtual machines). You cannot trust `/etc/mtab` content, go directly for `/proc/self/mounts`.

Same issue happens if CMSSW is compiled with ASan, in this case `realpath` was not doing a correct job for "`.`". Could be a bug in ASan interceptor.

### CODE REVIEW

**Comment by davidlt** [comment]
This also would be needed in 7_4_X.

**Comment by davidlt** [comment]
Tests are scheduled.

File: Utilities/StorageFactory/src/LocalFileSystem.cc [hunk 1]
```diff
@@ -212,6 +212,7 @@ LocalFileSystem::initFSInfo(void *arg)
 int
 LocalFileSystem::initFSList(void)
 {
+#define _PATH_MOUNTED_LINUX "/proc/self/mounts"
```
Comment (Dr15Jones) [inline Utilities/StorageFactory/src/LocalFileSystem.cc]: Change this to a ```const char * const PATH_MOUNTED_LINUX``` and declaring it only in the block it is needed.

**Comment by Dr15Jones** [comment]
@bbockelm @dan131riley do you guys have comments from a systems perspective?

**Comment by davidlt** [comment]
Modified. You can check the diff again.

**Comment by bbockelm** [comment]
This looks appropriate to me.  Use of /etc/mtab is a bit historical (this is old code) anyway - I think /proc/self/mounts is really the correct way going forward.

**Comment by blinkseb** [comment]
Hi there,

I can confirm that this patch fix the crash I observed on our remote UIs where /etc/mtab is not in sync with /proc/self/mounts.

Thanks!

**Comment by Dr15Jones** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**